### ISSUE TITLE
[1.10.2] Superfluous parameter in EntityRegistry#lookupModSpawn

### ISSUE THREAD

**Issue by Choonster** [MinecraftForge/MinecraftForge#3237]
The `boolean` parameter of `EntityRegistry#lookupModSpawn(Class<? extends Entity>, boolean)` can be replaced with a local variable. It's always assigned a value before its value is read, so the value passed to the method is ignored.

**Comment by mezz** [comment]
This is probably a bug with the logic, it looks like the boolean param is called with `false` and with `true` elsewhere, so they probably expect it to do something.

The fix should probably be to change this line:

``` java
keepLooking = (!Object.class.equals(localClazz));
```

to

``` java
keepLooking &= (!Object.class.equals(localClazz));
```

**Comment by Actuarius** [comment]
@mezz added labels [Bug]

**Comment by Jezza** [comment]
Fixed with #3346
